shared/middleware/helpers/auth_helpers.rb

```diff
@@ -35,7 +35,12 @@ def under_13?
 def get_user_sharing_disabled(user_id)
   user_properties = DASHBOARD_DB[:users].select(:properties).first(id: user_id)
   return false unless user_properties
-  parsed_properties = JSON.parse(user_properties[:properties])
+  get_sharing_disabled_from_properties(user_properties[:properties])
+end
+
+def get_sharing_disabled_from_properties(properties)
+  return false unless properties
+  parsed_properties = JSON.parse(properties)
   return false unless parsed_properties
```

Comment:
This line isn't needed. AFAIK `JSON.parse` will never return nil. It will either return parsed json as a hash, or raise an error:

```
irb(main):021:0> JSON.parse nil.to_json
JSON::ParserError: 784: unexpected token at 'null'

irb(main):023:0> JSON.parse nil
TypeError: no implicit conversion of nil into String

irb(main):024:0> JSON.parse nil.to_s
JSON::ParserError: A JSON text must at least contain two octets!

irb(main):026:0> JSON.parse ''
JSON::ParserError: A JSON text must at least contain two octets!

irb(main):028:0> JSON.parse '{}'
=> {}
```